### PR TITLE
Set log stream natively on Android 

### PR DESCRIPTION
**Pull request by erdemyerebasmaz**
Node log streams is set natively on Android and communicated to app via an EventChannel. 

Relies on Breez SDK changes:
- https://github.com/breez/breez-sdk/pull/727

#### Other changes:
- Call startForeground when service starts
  - An issue faced because app & service share the same process & EventChannel has to use main thread when communicating to app
```
E/AndroidRuntime( 3300): FATAL EXCEPTION: main
E/AndroidRuntime( 3300): Process: com.cBreez.client.debug, PID: 3300
E/AndroidRuntime( 3300): android.app.RemoteServiceException: Context.startForegroundService() did not then call Service.startForeground(): ServiceRecord{839d82d u0 com.cBreez.client.debug/com.cBreez.client.BreezForegroundService}
E/AndroidRuntime( 3300): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2540)
E/AndroidRuntime( 3300): 	at android.os.Handler.dispatchMessage(Handler.java:110)
E/AndroidRuntime( 3300): 	at android.os.Looper.loop(Looper.java:219)
E/AndroidRuntime( 3300): 	at android.app.ActivityThread.main(ActivityThread.java:8668)
E/AndroidRuntime( 3300): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 3300): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
E/AndroidRuntime( 3300): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1109)
I/Process ( 3300): Sending signal. PID: 3300 SIG: 9
```

#### TODO:
- Sever the EventChannel connection when app is killed
  - `FlutterJNI( 5292): Tried to send a platform message to Flutter, but FlutterJNI was detached from native C++. Could not send.`

### CODE REVIEW

File: android/app/src/main/kotlin/com/cBreez/client/BreezForegroundService.kt [hunk 1]
```diff
@@ -82,6 +82,10 @@ class BreezForegroundService : ForegroundService, Service() {
         val intentDetails = "[ intent=$intent, flag=$flags, startId=$startId ]"
         Logger.tag(TAG).debug { "Start Breez foreground service from intent $intentDetails" }
 
+        // Display foreground service notification when connecting for the first time
+        val notification = notifyForegroundService(applicationContext)
+        startForeground(NOTIFICATION_ID_FOREGROUND_SERVICE, notification)
```
[the remark this answers is not in the document]
Comment (erdemyerebasmaz) [reply android/app/src/main/kotlin/com/cBreez/client/BreezForegroundService.kt:87]: We had encountered an error on live build: 
`Context.startForegroundService() did not then call Service.startForeground()`
due to background execution limits on Android.

> The new Context.startForegroundService() method starts a foreground service. The system allows apps to call Context.startForegroundService() even while the app is in the background. However, the app must call that service's [startForeground()](https://developer.android.com/reference/android/app/Service#startForeground(int,%20android.app.Notification)) method within five seconds after the service is created.

[Android 8.0 Behavior Changes > Background execution limits](https://developer.android.com/about/versions/oreo/android-8.0-changes.html#back-all)

**Review by roeierez — APPROVED** [review]
LGTM

**Comment by erdemyerebasmaz** [comment]
@roeierez Thanks for the review. There's one remaining issue where we print double logs on Android as we're subscribed to the log stream from the Flutter app & natively on Android.

**Comment by erdemyerebasmaz** [comment]
> @roeierez Thanks for the review! There's one remaining issue where we print double logs on Android as we're subscribed to the log stream from the Flutter app & natively on Android.
> 
> Partially addressed with: [Subscribe to node log stream on creating Breez foreground service](https://github.com/breez/c-breez/pull/762/commits/b7674eca04669f7a06951b16499a2e346b23faf1)
> 
> I believe the correct approach here is to make sure we only subscribe when the app is killed.

Decided to disable printing tinylog logs to logcat as a temporary solution.

**Pull request merged**